File: app/src/main/java/code/name/monkey/retromusic/glide/palette/BitmapPaletteTarget.java
/*
 * Copyright (c) 2019 Hemanth Savarala.
 *
 * Licensed under the GNU General Public License v3
 *
 * This is free software: you can redistribute it and/or modify it under
 * the terms of the GNU General Public License as published by
 *  the Free Software Foundation either version 3 of the License, or (at your option) any later version.
 *
 * This software is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
 * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
 * See the GNU General Public License for more details.
 */

package code.name.monkey.retromusic.glide.palette;

import android.widget.ImageView;
import com.bumptech.glide.request.target.ImageViewTarget;

public class BitmapPaletteTarget extends ImageViewTarget<BitmapPaletteWrapper> {
  public BitmapPaletteTarget(ImageView view) {
    super(view);
  }

  @Override
  protected void setResource(BitmapPaletteWrapper bitmapPaletteWrapper) {
    view.setImageBitmap(bitmapPaletteWrapper.getBitmap());
  }
}

File: app/src/main/java/code/name/monkey/retromusic/glide/palette/BitmapPaletteTarget.kt
package code.name.monkey.retromusic.glide.palette

import android.widget.ImageView

import com.bumptech.glide.request.target.ImageViewTarget

open class BitmapPaletteTarget(view: ImageView) : ImageViewTarget<BitmapPaletteWrapper>(view) {

    override fun setResource(bitmapPaletteWrapper: BitmapPaletteWrapper?) {
        if (bitmapPaletteWrapper != null) {
            view.setImageBitmap(bitmapPaletteWrapper.bitmap)
        }
    }
}